ssl/ssl_init.c

```diff
@@ -61,6 +66,30 @@ DEFINE_RUN_ONCE_STATIC_ALT(ossl_init_no_load_ssl_strings,
     return 1;
 }
 
+static void pin_library(void)
+{
```

Comment:
This code is now here, in OPENSSL_atexit() and in ossl_init_load_crypto_nodelete().
Maybe create a `int DSO_pinbyaddr(void *addr)` in [dso_lib.c](https://github.com/openssl/openssl/tree/master/crypto/dso/dso_lib.c) and [dso.h](https://github.com/openssl/openssl/blob/master/include/internal/dso.h) and use it for all these places?